src/main/app/idam/authorizationMiddleware.ts

```diff
@@ -0,0 +1,65 @@
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+
+import { JwtExtractor } from 'idam/jwtExtractor';
+import { IdamClient } from 'idam/idamClient';
+import { User } from 'idam/user';
+import { Logger } from '@hmcts/nodejs-logging';
+
+import config from 'config';
+
+const sessionCookieName = config.get<string>('session.cookieName');
+const logger = Logger.getLogger('middleware/authorization');
+
+export function hasTokenExpired (err): boolean {
+  return (err.statusCode === HttpStatus.FORBIDDEN || err.statusCode === HttpStatus.UNAUTHORIZED);
+}
+
+export class AuthorizationMiddleware {
+
+  static requestHandler (
+    requiredRoles: string[],
+    accessDeniedCallback: (req: express.Request, res: express.Response) => void,
```

Comment:
Why have this passed in as a callback? Redirect is always going to be login no?

Reply:
not sure if i understand this!

Reply:
The method takes in `accessDeniedCallback` but this is always going to be login? Should we not just have a private method within this module to handle the redirect?

Reply:
moved it to private